packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -850,19 +866,19 @@ class FlutterVmService {
     }
 
     final Completer<vm_service.IsolateRef> extensionAdded = Completer<vm_service.IsolateRef>();
-    StreamSubscription<vm_service.Event> isolateEvents;
+    late final StreamSubscription<vm_service.Event> isolateEvents;
     isolateEvents = service.onIsolateEvent.listen((vm_service.Event event) {
       if (event.kind == vm_service.EventKind.kServiceExtensionAdded
           && event.extensionRPC == extensionName) {
-        isolateEvents?.cancel();
+        isolateEvents.cancel();
         extensionAdded.complete(event.isolate);
       }
     });
 
     try {
       final List<vm_service.IsolateRef> refs = await _getIsolateRefs();
       for (final vm_service.IsolateRef ref in refs) {
-        final vm_service.Isolate isolate = await getIsolateOrNull(ref.id);
+        final vm_service.Isolate? isolate = await getIsolateOrNull(ref.id!);
```

Comment:
Can `getIsolateOrNull` return `null` in the case of a null `id` instead? Or is that more of an error case?

Reply:
Hmm, I'm not sure. The definition of `vm_service.IsolateRef` defines it as nullable, but I'm not sure what would lead to this state. I'm assuming on master now, if we call `getIsolateOrNull(null)` it will lead to an error calling `getIsolate` from the vmservice.

I don't have a strong opinion either way, but I think I would slightly prefer to get a null check error then for the code to silently keep moving. However I'd defer to you if you think we should just return null in that case.